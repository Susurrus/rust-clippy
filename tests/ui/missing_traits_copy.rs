#![feature(untagged_unions)]
#![allow(dead_code)]
#![allow(clippy::all)]
#![warn(clippy::missing_copy_implementations)]

pub enum A {}

#[derive(Clone, Copy)]
pub enum B {}

pub enum C {}

impl Copy for C {}
impl Clone for C {
    fn clone(&self) -> C { *self }
}

pub struct Foo;

#[derive(Clone, Copy)]
pub struct Bar;

pub struct Baz;

impl Copy for Baz {}
impl Clone for Baz {
    fn clone(&self) -> Baz { *self }
}

struct PrivateStruct;

enum PrivateEnum {}

struct GenericType<T>(T);

fn main() {}
