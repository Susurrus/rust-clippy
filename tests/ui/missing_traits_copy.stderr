error: type does not implement `Copy`; consider adding #[derive(Copy)] or a manual implementation
  --> $DIR/missing_traits_copy.rs:6:1
   |
LL | pub enum A {}
   | ^^^^^^^^^^^^^
   |
   = note: `-D clippy::missing-copy-implementations` implied by `-D warnings`

error: type does not implement `Copy`; consider adding #[derive(Copy)] or a manual implementation
  --> $DIR/missing_traits_copy.rs:18:1
   |
LL | pub struct Foo;
   | ^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

